File: 백준/Silver/1149. RGB거리/RGB거리.py
N = int(input())
colors = [[0, 0, 0]] + [list(map(int, input().split())) for _ in range(N)]

cost = [[0, 0, 0] for _ in range(N+1)]
cost[1] = colors[1][:]

for i in range(2, N+1):
    cost[i][0] = colors[i][0] + min(cost[i-1][1], cost[i-1][2])
    cost[i][1] = colors[i][1] + min(cost[i-1][0], cost[i-1][2])
    cost[i][2] = colors[i][2] + min(cost[i-1][0], cost[i-1][1])

print(min(cost[N]))

File: 백준/Silver/1149. RGB거리/RGB거리.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main
{
    public static void main(String[] args) throws Exception
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] cost = new int[1001][4];
        for (int i = 1; i <= n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            for (int j = 1; j <= 3; j++) {
                cost[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        int[][] d = new int[1001][4]; // 누적 점수, 어떤 집인지 
        d[1][1] = cost[1][1];
        d[1][2] = cost[1][2];
        d[1][3] = cost[1][3];
        for (int i = 2; i <= n; i++) {
            d[i][1] = Math.min(d[i-1][2], d[i-1][3]) + cost[i][1];
            d[i][2] = Math.min(d[i-1][1], d[i-1][3]) + cost[i][2];
            d[i][3] = Math.min(d[i-1][1], d[i-1][2]) + cost[i][3];
        }
        
        int answer = Integer.MAX_VALUE;
        for (int i = 1; i <= 3; i++) {
            answer = Math.min(answer, d[n][i]);
        }
        
        System.out.println(answer);
    }
}